From a941869c2e3de9cf53e706b8547863eb2580820d Mon Sep 17 00:00:00 2001
From: Hao Dong <spdonghao@google.com>
Date: Fri, 11 Oct 2024 03:56:57 +0000
Subject: [PATCH] Set class name when createConfirmDeviceCredentialIntent() API
 is used.

Flag: EXEMPT bugfix
Bug: 339532378
Test: manual test
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:1ce2fd4cef984ee7d2392d9abeca7560e17f8917)
Merged-In: I762f280def79d428ed9cf119ace5ccc2b6be2560
Change-Id: I762f280def79d428ed9cf119ace5ccc2b6be2560
---
 src/com/android/settings/password/BiometricFragment.java | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/com/android/settings/password/BiometricFragment.java b/src/com/android/settings/password/BiometricFragment.java
index 4ad04a1a903..97ed8e5fea7 100644
--- a/src/com/android/settings/password/BiometricFragment.java
+++ b/src/com/android/settings/password/BiometricFragment.java
@@ -18,6 +18,7 @@ package com.android.settings.password;
 
 import android.app.settings.SettingsEnums;
 import android.hardware.biometrics.BiometricPrompt;
+import android.hardware.biometrics.BiometricPrompt.Builder;
 import android.hardware.biometrics.BiometricPrompt.AuthenticationCallback;
 import android.hardware.biometrics.BiometricPrompt.AuthenticationResult;
 import android.hardware.biometrics.PromptInfo;
@@ -142,6 +143,7 @@ public class BiometricFragment extends InstrumentedFragment {
                         promptInfo.isDisallowBiometricsIfPolicyExists())
                 .setReceiveSystemEvents(true)
                 .setAllowBackgroundAuthentication(true)
+                .setClassNameIfItIsConfirmDeviceCredentialActivity()
                 .build();
     }
 
-- 
2.34.1

